### PR TITLE
[SW-82] Sign transaction popups

### PR DESCRIPTION
**Pull request by pstachel**

### CODE REVIEW

File: src/popup/router/components/Modals/ConfirmTransactionSign.vue [hunk 1]
```diff
@@ -130,45 +162,146 @@ export default {
...
+      }
+    }
+  },
+  methods: {
+    getTokens(txParams) {
+      if (!this.isDex) return [this.singleToken];
+      const resolver = transactionTokenInfoResolvers[camelCase(txParams.function)];
+      if (!resolver) return [];
+      const tokens = resolver(
+        { tx: { ...txParams, ...this.transaction } }, this.availableTokens,
```
Comment (CedrikNikita) [inline src/popup/router/components/Modals/ConfirmTransactionSign.vue:246]: **fix**: there was not enough info for transactions: Allowance and AE related
image
image

File: src/popup/router/components/Modals/ConfirmTransactionSign.vue [hunk 2]
```diff
@@ -130,45 +162,146 @@ export default {
...
+    completeTransaction() {
+      return { ...this.transaction, function: this.txFunction };
+    },
+  },
+  async mounted() {
+    if (this.transaction.contractId) {
+      try {
+        this.loading = true;
+        setTimeout(() => { this.loading = false; }, 20000);
+        await this.$watchUntilTruly(() => this.sdk);
```
Comment (CedrikNikita) [inline src/popup/router/components/Modals/ConfirmTransactionSign.vue:219]: **fix** error on initing the popup (waiting for `sdk` being trully to get a `compilerApi`), was not loading any data at all
`TypeError: Cannot read properties of null (reading 'compilerApi')`

File: src/popup/router/components/Modals/ConfirmTransactionSign.vue [hunk 3]
```diff
@@ -3,9 +3,21 @@
     full-screen
     class="confirm-transaction-sign"
     data-cy="popup-aex2"
+    absolute-footer
   >
-    <TransactionOverview :tx="transaction" />
-
+    <TransactionOverview :tx="completeTransaction" />
```
Comment (CedrikNikita) [inline src/popup/router/components/Modals/ConfirmTransactionSign.vue:8]: **fix**: new labels were not shown

File: src/popup/router/components/Modals/ConfirmTransactionSign.vue [hunk 4]
```diff
@@ -130,45 +162,146 @@ export default {
...
+      return { ...this.transaction, function: this.txFunction };
+    },
+  },
+  async mounted() {
+    if (this.transaction.contractId) {
+      try {
+        this.loading = true;
+        setTimeout(() => { this.loading = false; }, 20000);
+        await this.$watchUntilTruly(() => this.sdk);
+        const { bytecode } = await fetchJson(`${this.activeNetwork.url}/v3/contracts/${this.transaction.contractId}/code`);
```
Comment (CedrikNikita) [inline src/popup/router/components/Modals/ConfirmTransactionSign.vue:220]: **fix**: additional information was loading only for the testnet

File: src/popup/router/components/Modals/ConfirmTransactionSign.vue [hunk 5]
```diff
@@ -130,45 +162,146 @@ export default {
...
+      if (['swap_tokens_for_exact_tokens', 'swap_tokens_for_exact_ae', 'swap_ae_for_exact_tokens'].includes(this.txFunction)) return 'minReceived';
+      if (['swap_exact_tokens_for_tokens', 'swap_exact_ae_for_tokens', 'swap_exact_tokens_for_ae'].includes(this.txFunction)) return 'maxSpent';
+      return '';
+    },
+    completeTransaction() {
+      return { ...this.transaction, function: this.txFunction };
+    },
+  },
+  async mounted() {
+    if (this.transaction.contractId) {
```
Comment (CedrikNikita) [inline src/popup/router/components/Modals/ConfirmTransactionSign.vue:215]: **fix**: do not load additional info for a none conractCall transaction

**Review by CedrikNikita — COMMENTED** [review]
- **fix** error on initing the popup (waiting for sdk being trully to get a compilerApi), was **not** loading any data at all
`TypeError: Cannot read properties of null (reading 'compilerApi')`
- remove unused getters
- **fix** wrong showing of which token is `from` and which is `to` for the swap
- **fix**: there was not enough info for transactions: Allowance and AE related
- **fix**: new labels were not shown
- **fix**: additional information was loading only for the testnet
- **fix**: do not load additional info for a none contractCall transaction
- **fix**: do not show pool label for the allowance transactions

File: src/popup/router/components/Modals/ConfirmTransactionSign.vue [hunk 6]
```diff
@@ -130,45 +162,146 @@ export default {
...
+      try {
+        this.loading = true;
+        setTimeout(() => { this.loading = false; }, 20000);
+        await this.$watchUntilTruly(() => this.sdk);
+        const { bytecode } = await fetchJson(`${this.activeNetwork.url}/v3/contracts/${this.transaction.contractId}/code`);
```
Comment (ifaouibadi) [inline src/popup/router/components/Modals/ConfirmTransactionSign.vue:220]:
```suggestion
        const { bytecode } = await this.sdk.;
```

File: src/popup/router/components/Overview.vue [hunk 7]
```diff
@@ -90,18 +109,36 @@ export default {
...
+    txFunction: { type: String, default: null },
     sender: { type: Object, required: true },
     recipient: { type: Object, required: true },
   },
+  computed: {
+    isNewUI() {
+      return !!this.$route.meta.newUI;
+    },
+    getTxType() {
+      return this.$te(`transaction.dexType.${getDexTransactionTag[this.txFunction] || this.txFunction}`)
```
Comment (ifaouibadi) [inline src/popup/router/components/Overview.vue:125]:
```suggestion
      return this.$t(`transaction.dexType.${getDexTransactionTag[this.txFunction] || this.txFunction}`)
```

File: src/popup/router/pages/TransactionDetails.vue [hunk 8]
```diff
@@ -201,6 +201,7 @@ export default {
       'getTxTipUrl',
       'getExplorerPath',
       'isTxAex9',
+      'getTxType',
```
Comment (ifaouibadi) [inline src/popup/router/pages/TransactionDetails.vue:204]:
```suggestion
     
```
since it's not used

**Review by ifaouibadi — CHANGES_REQUESTED** [review]

File: src/popup/router/components/Overview.vue (hunk 7 above)
Comment (CedrikNikita) [reply src/popup/router/components/Overview.vue:125]: https://github.com/aeternity/superhero-wallet/pull/1492#discussion_r933164306

**Comment by CedrikNikita** [comment]
Wrong token amount is shown.
Wrong token

**Comment by CedrikNikita** [comment]
image

File: src/popup/router/components/TransactionDetailsPoolTokenRow.vue [hunk 9]
```diff
@@ -0,0 +1,112 @@
+<template>
```
Comment (CedrikNikita) [inline src/popup/router/components/TransactionDetailsPoolTokenRow.vue:1]: Don't see a good reason to extract this component.

**Pull request merged**